vscode/src/diffView/register.ts

```diff
@@ -0,0 +1,84 @@
+import * as vscode from "vscode";
+import { FileItem, KonveyorFileModel, KonveyorTreeDataProvider } from "./fileModel";
+import { MemFS } from "./fileSystemProvider";
+import { testLocations, testPaths, writeResponse } from "./mock";
+import fs from "fs";
+import path from "path";
+import { Navigation } from "./navigation";
+
+export function registerDiffView(context: vscode.ExtensionContext): void {
+  const memFs = new MemFS();
+  context.subscriptions.push(
+    vscode.workspace.registerFileSystemProvider("konveyorMemFs", memFs, {
+      isCaseSensitive: true,
+    }),
+  );
+  const changedFiles = Object.fromEntries(
+    testPaths.map((fsPath) => [
+      fsPath,
+      fs.readFileSync(path.join(context.extensionPath, "src", "diffView", "resolutions", fsPath), {
+        encoding: "utf8",
+      }),
+    ]),
+  );
+  writeResponse(changedFiles, memFs);
+  const model = new KonveyorFileModel(testLocations);
+  const provider = new KonveyorTreeDataProvider(model);
+  vscode.window.registerTreeDataProvider("konveyor.diffView", provider);
+  const treeView = vscode.window.createTreeView<unknown>("konveyor.diffView", {
+    treeDataProvider: provider,
+    showCollapseAll: true,
+  });
+  new Navigation(treeView, model);
+
+  treeView.message = model.message;
+  context.subscriptions.push(treeView);
+
+  provider.onDidChangeTreeData(() => {
+    treeView.message = model.message;
+  });
+
+  context.subscriptions.push(
+    vscode.commands.registerCommand("konveyor.diffView.applyAll", applyAll),
+    vscode.commands.registerCommand("konveyor.diffView.revertAll", revertAll),
+    vscode.commands.registerCommand("konveyor.diffView.applyFile", applyFile),
+    vscode.commands.registerCommand("konveyor.diffView.revertFile", revertFile),
+    vscode.commands.registerCommand("konveyor.diffView.copyDiff", copyDiff),
+    vscode.commands.registerCommand("konveyor.diffView.copyPath", copyPath),
+  );
+}
+
+function applyAll() {
+  vscode.window.showInformationMessage(`[TODO] Apply all resolutions`);
+}
+
+function revertAll() {
+  vscode.window.showInformationMessage(`[TODO] Discard all local changes to resolutions`);
+}
+
+function applyFile(item: FileItem) {
+  item.apply();
+}
+
+function revertFile(item: FileItem) {
+  item.revert();
+}
+async function copyDiff(item: KonveyorFileModel | FileItem | unknown) {
```

Comment:
I am picking this one as an example to support the point I made in [this comment](https://github.com/konveyor/editor-extensions/pull/79/files#r1815412315).

Could you please articulate for me, in detail, why this function could not be defined in [the commandsMap of commands.ts where it would also be registered](https://github.com/konveyor/editor-extensions/blob/65272448356155f4089c64cb8c3dfca2ee68290e/vscode/src/commands.ts#L22C7-L22C18).

Reply:
@djzager 
Technically there are no problems. However let me extend [the answer above](https://github.com/konveyor/editor-extensions/pull/79#issuecomment-2432592967):
1. all commands are already listed in `package.json` -  note that only string IDs are used so there is no dependency on the code. Command map duplicates this registry BUT
2. differently then `packaged.json` it works on actual code which brings side effects:
   1.   size - the file is already >300 lines and would need to keep growing forever
   2.  imports - the file needs to import code from all modules/units
   3.  volatility - the file is likely to be updated by unrelated commits

If you feel more confident to register commands in one place then I would recommend that each module/unit should export the their small command maps that would get merged before registration.

Reply:
@djzager 
Note that `command.ts` is the most frequently changed production source code with volatility rate the same as `yarn.lock` . Below 10 most volatile files in Continue's vscode extension (`git log --name-only  --pretty=format: | sort | uniq -c | sort -nr | grep vscode`)
```
    508 extensions/vscode/package.json
    456 extensions/vscode/package-lock.json
    223 extensions/vscode/config_schema.json
    180 extensions/vscode/continue_rc_schema.json
    126 extensions/vscode/src/debugPanel.ts
    118 extensions/vscode/yarn.lock
    118 extensions/vscode/src/commands.ts
    117 extensions/vscode/src/ideProtocol.ts
    103 extensions/vscode/scripts/prepackage.js
     76 extensions/vscode/src/webviewProtocol.ts
```

Reply:
It is extremely helpful to me to see these metrics. I think the increased volatility is a fair price to pay to make navigating our codebase at this stage as easy as possible. I'm hopeful that what we discussed in #87 is a reasonable/meaningful compromise that mitigates some of the risk you have identified.